copy_edit.py

```diff
@@ -33,7 +33,7 @@ def copy(self, edit):
 			if len(s):
 				new_sel_strings.append((self.view.substr(s), False))
 			elif copy_with_empty_sel:
-				new_sel_strings.append((self.view.substr(self.view.full_line(s)), True))
+				new_sel_strings.append((self.view.substr(self.view.line(s)) + "\n", True))
```

Comment:
What bug does this fix? Why would you want to manually add a newline?

Reply:
It fixes case when selection is empty and when caret is placed at the last line of file and when there is no newline at end of file. Do you use copy with empty selection? If not, then you wouldn't understand this fix.